src/projectApplications.js

```diff
@@ -0,0 +1,137 @@
+const fs = require('fs');
+const path = require('path');
+const {google} = require('googleapis');
+
+const SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly'];
+const ALL_QUARTERS_ID = '16vmuNiN-8mij8EoEIwvQ_wde9DgF_0IWXfT_95SHfcc';
+
+const numOfQuarters = 14;
+
+// Load client secrets from a service account
+fs.readFile(path.resolve(__dirname, '../credentials.json'), (err, content) => {
+  authorize(JSON.parse(content), main);
+});
+
+function authorize(credentials, callback) {
+  const jwt = new google.auth.JWT(credentials.client_email, null, credentials.private_key, SCOPES);
+  callback(jwt);
+}
+
+function main(auth) {
+  process.stdout.write('Parsing project sheet .. ');
+  readProjectMemberSheet(auth);
+}
+
+function readProjectMemberSheet(auth) {
+  const sheets = google.sheets({version: 'v4', auth});
+  const endQuarter = ['Winter', '2021'];
+  const startQuarter = ['Fall', '2016'];
+  const quarters = ['Winter', 'Spring', 'Fall'];
+  let current = startQuarter;
+  let masterDict = [];
+  let count = 0;
+
+  
+  while(current[1] < endQuarter[1] ||
+    (current[1] == endQuarter[1] && quarters.indexOf(current[0]) <= quarters.indexOf(endQuarter[0]))){
+    
+    const currentQuarter = current[0];
+    const currentYear = current[1];
+    // Run on current quarter
+    sheets.spreadsheets.values.get({
+      spreadsheetId: ALL_QUARTERS_ID,
+      range: `${current[0]} ${current[1]}!A:Z`,
+    }).then((response) => {
+      const info = formatProjectMemberResponse(response);
+      // console.log(info);
```

Comment:
let's follow the ideas of the `pleadApplication.js`

the idea is:
1. after each quarter, we append all the information we get onto `masterDict`
2. every quarter, we call `batchedJsonCallback`
3. On the very last quarter, when we call `batchedJsonCallback` we are guaranteed all our information is there (we wont run into race conditions because we are just adding onto an area)
4. Once we are on the last quarter, we can do all the processing that you are doing here